### DISCUSSION TITLE
[Idea] Provide the ability to pass the current value in a filter as a parameter to a macro "pipe"

### DISCUSSION THREAD

**Discussion by AnthonyMuscio** [TiddlyWiki/TiddlyWiki5#6890] [Ideas]
As discussed here [The format Operator](https://talk.tiddlywiki.org/t/the-format-operator/4220/3?u=tw_tones) and prior to that #5073 It has dawned on me that if we could pipe the intermediate values in a filter, to a macro in the filter, it would provide powerful functionality.

* I mention it now because it may be important to consider and include this in the #6666 and related upcoming changes.

In effect I wonder if we could create a pipe operator which would pass the current filter value into a macro using a named parameter as described below;

`...pipe:paramname<macroname otherparam:"value">....`
* That is the pipe operator will set "paramname" to the current filter value before calling `macroname`
* paramname could default to `filter-value` for which the macroname would need to include in its definition to make use of it.
  * Allowing   `...pipe<macroname>....`

**Below is from my talk.tiddlywiki.org post**

Actually with the fact we can now use parameters in macros, within filters and the possibilities of defining our own operators with "Parametrised transclusions" on the horizon this is only getting easier to do.

I see that I had shared a macro to do "number formatting" on 24 Nov 2020

However a question that arises is;
* What if our macros used in a filter, could receive as a parameter the current value within the filter?.
 eg "`[all[current]get[amount]<format-number p:3>]`"
or if necessary `[all[current]get[amount]pipe:n<format-number p:3>]` format number with 3 places.
where a new pipe operator names the parameter to use, in this case "n"
* Basically this would be providing a "pipe" to allow a value to be piped into the macro, as it stands the output is passed back into the filter.

What do you think?
